### PR TITLE
Extract dataset management methods to a separate interface

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

**Comment by anew** [comment]
why is this needed? Jira?

File: cdap-api/src/main/java/co/cask/cdap/api/Admin.java [hunk 1]
```diff
@@ -17,80 +17,14 @@
...
 @Beta
-public interface Admin extends SecureStoreManager, MessagingAdmin {
-  /**
-   * Check whether a dataset exists in the current namespace.
-   * @param name the name of the dataset
-   * @return whether a dataset of that name exists
-   * @throws DatasetManagementException for any issues encountered in the dataset system
-   */
-  boolean datasetExists(String name) throws DatasetManagementException;
+public interface Admin extends DatasetManager, SecureStoreManager, MessagingAdmin {
```
Comment (anew) [inline cdap-api/src/main/java/co/cask/cdap/api/Admin.java:28]: why are some of these interfaces called "Manager" and some "Admin"?

Comment (chtyim) [reply cdap-api/src/main/java/co/cask/cdap/api/Admin.java:28]: unfortunately there is already a `DatasetAdmin` in api, which is for administration of single dataset type.

Comment (chtyim) [reply cdap-api/src/main/java/co/cask/cdap/api/Admin.java:28]: For the `SecureStoreManager`, not sure.

**Comment by chtyim** [comment]
The refactoring allows more flexible combination of using specific admin functionalities.

**Comment by anew** [comment]
LGTM

**Comment by chtyim** [comment]
Building http://builds.cask.co/browse/CDAP-DUT5399-1

**Comment by chtyim** [comment]
Build passed

**Pull request merged**